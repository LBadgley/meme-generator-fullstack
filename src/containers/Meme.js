import React, { PureComponent } from 'react';
import DisplayView from '../components/UserView';
import TextPicker from '../components/TextComponent';
import ImagePicker from '../components/ImageComponent';

export default class Meme extends PureComponent {
  state = {
    topText: '',
    image: '',
    bottomText: ''
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { topText, bottomText, image } = this.state;

    return (
      <>
        <DisplayView topText={topText} bottomText={bottomText} image={image} handleChange={this.handleChange}/>
        <ImagePicker image={image} handleChange={this.handleChange} />
        <TextPicker topText={topText} bottomText={bottomText} handleChange={this.handleChange}/>
      </>
    );
  }
}
