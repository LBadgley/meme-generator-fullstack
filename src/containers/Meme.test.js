import React from 'react';
import { shallow } from 'enzyme';
import Meme from './Meme';

describe('meme tests', () => {
  it('renders the Meme', () => {
    const wrapper = shallow(<Meme />);
    expect(wrapper).toMatchSnapshot();
  });
});
