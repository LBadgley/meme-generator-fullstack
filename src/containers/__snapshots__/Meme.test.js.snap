// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meme tests renders the Meme 1`] = `
<Fragment>
  <DisplayView
    bottomText=""
    handleChange={[Function]}
    image=""
    topText=""
  />
  <ImagePicker
    handleChange={[Function]}
    image=""
  />
  <TextPicker
    bottomText=""
    handleChange={[Function]}
    topText=""
  />
</Fragment>
`;
